kolibri/core/public/api.py

```diff
@@ -324,3 +327,35 @@ def create(self, request):
 
     def update(self, request, pk=None):
         return self.check_queue(request, pk=pk)
+
+
+class FacilitySearchUsernameViewSet(ReadOnlyValuesViewset):
```

Comment:
I think you only want to serve list requests here, so probably better to inherit from `BaseValuesViewset` - otherwise this endpoint is exposing a detail endpoint as well, with no permission checks, and the ability to look up any user by id.

Reply:
Of course, I'm afraid I don't remember all the possible viewsets we have in the kolibri code and I picked the first I saw that did the work. Looking at it it's clear using `BaseValuesViewset` limits perfectly the scope.